WordPress/gutenberg
# [Mobile] - Remove `react-native-hr`

**Related PRs:**
- https://github.com/wordpress-mobile/gutenberg-mobile/pull/5550

## What?
This PR removes the `react-native-hr` library in favor of having our own component.

## Why?
To reduce the usage of unnecessary libraries and also of the current forks the project uses.

## How?
By implementing the `HR` mobile component that also has a `HorizontalRule` wrapper.

It adds a new functionality to set a `color` with this, now we'll able to support customizing the color of the Separator block.

## Testing Instructions

On both platforms:

### Test case 1

- Add a `Page break` block
- Add a `More ` block
- Add a `Separator` block
- **Expect** the three blocks to show as expected (no visual regressions)

### Test case 2

- Add a `Separator` block
- Open the Block settings
- Set a color
- Expect to see the line with the selected color.

### Testing Instructions for Keyboard
N/A

## Screenshots or screencast 

Before iOS | After iOS
-|-
<kbd><img src="https://user-images.githubusercontent.com/4885740/224328422-dfba045a-72fc-44d2-b59f-69f7fbcdc5c7.png" width="200" /></kbd> | <kbd><img src="https://user-images.githubusercontent.com/4885740/224328297-a7176660-dbb5-4d92-9328-53fc16461273.png" width="200" /></kbd>

Before Android | After Android
-|-
<kbd><img src="https://user-images.githubusercontent.com/4885740/224328383-6d9e8d6d-49c2-4260-ac9c-61919e4d9531.png" width="200" /></kbd> | <kbd><img src="https://user-images.githubusercontent.com/4885740/224328334-409fad62-741a-4786-aaa6-3493fdcbde21.png" width="200" /></kbd>
